### PR TITLE
Tools

### LINKED ISSUE #88: set vvisp config
**Subject**
vvisp

**Detail**
set vvisp configuration

### PR DESCRIPTION
**Pull request by junbeomlee**
## Related Issue

resolve: #88 #89 

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] Test case
- [x] End of Work

### CODE REVIEW

File: contracts/ethereum/contractApis/back/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  apis['GhostAuction'] = require('./js/GhostAuction.js');
+  apis['GhostBase'] = require('./js/GhostBase.js');
+  apis['GhostFactory'] = require('./js/GhostFactory.js');
+  apis['GhostOwnership'] = require('./js/GhostOwnership.js');
+  apis['SaleAuction'] = require('./js/SaleAuction.js');
+  apis['SpecialAuction'] = require('./js/SpecialAuction.js');
+  return apis;
+}
+
+const setters = {
```
Comment (hihiboss) [inline contracts/ethereum/contractApis/back/index.js:37]: wow...

File: contracts/ethereum/tools/lib/factories/contractFactory.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export default class ContractFactory {
+  private defaultPrivateKey!: string;
+  private readonly endpoint!: string;
+
+  constructor(privateKey: string, endpoint: string) {
+    this.defaultPrivateKey = privateKey;
+    this.endpoint = endpoint;
+  }
+
+  public createGhostContract(
```
Comment (hihiboss) [inline contracts/ethereum/tools/lib/factories/contractFactory.ts]: Its name is confused! `createGhostFactory` looks better in my thought.

File: contracts/ethereum/tools/lib/hatchoutContractFactory.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  public createDefaultGhostFactoryContract(): GhostFactoryContract {
+    return this.contractFactory
+      .createGhostContract(
+        this.vvispStateSupplier.loadContractAddress(
+          "GhostFactory",
+        )
+      );
+  }
+
+  public createGhostAuctionContractFromVvispState(): GhostAuctionContract {
```
Comment (hihiboss) [inline contracts/ethereum/tools/lib/hatchoutContractFactory.ts]: Why are the names of `createDefaultGhostFactoryContract` and `createGhostAuctionContractFromVvispstate` different?

File: contracts/ethereum/tools/lib/utils/utils.ts [hunk 4]
```diff
@@ -8,17 +9,34 @@ export default class Utils {
...
+    if (defaultOwner === undefined) {
       throw new Error('at least one wallet must exist');
     }
-    const hash = await this.web3.utils.soliditySha3(gene, defaultOwner);
+
+    return defaultOwner
+  }
+
+  public async createGeneSignature(gene: Mixed): Promise<string> {
+    return await this.createSignature(gene, await this.loadOwner())
```
Comment (hihiboss) [inline contracts/ethereum/tools/lib/utils/utils.ts]: Why did `this.loadOwner()` used as a parameter??

In `createSignature()`, it was already signed with `deafaultOwner`.

File: contracts/ethereum/tools/bin/createSaleAuction.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    const hatchOutContract: HatchOutContract = hatchOut
+      .factory
+      .createDefaultHatchOutContract();
+
+    const receipt = await hatchOutContract
+      .methods
+      .createSaleAuction(
+        tokenId,
+        buyer,
+        (await hatchOut.utils.toBN('1000')).toString()
```
Comment (hihiboss) [inline contracts/ethereum/tools/bin/createSaleAuction.ts]: What does `1000` mean?

File: contracts/ethereum/packages/hatchout-cli/src/commands/ghost/list.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import {Command, flags} from '@oclif/command'
+import HatchOut from "../../../../hatchout-core/hatchOut";
+
+export default class List extends Command {
+  static description = 'describe the command here';
+
+  static examples = [
```
Comment (hihiboss) [inline contracts/ethereum/packages/hatchout-cli/src/commands/ghost/list.ts:7]: This is a file for `list`, but example is explaining `hello`! Is it right?

**Review by hihiboss — COMMENTED** [review]
LGTM 👍
But plz check my comments and answer about them!

File: contracts/ethereum/tools/lib/factories/contractFactory.ts (hunk 2 above)
Comment (junbeomlee) [reply contracts/ethereum/tools/lib/factories/contractFactory.ts]: `ContractFactory` has the role that creates the contract.
So createXXXConctract is not confused!

File: contracts/ethereum/tools/lib/hatchoutContractFactory.ts (hunk 3 above)
Comment (junbeomlee) [reply contracts/ethereum/tools/lib/hatchoutContractFactory.ts]: renamed it!

File: contracts/ethereum/tools/bin/createSaleAuction.ts (hunk 5 above)
Comment (junbeomlee) [reply contracts/ethereum/tools/bin/createSaleAuction.ts]: removed!

File: contracts/ethereum/tools/lib/utils/utils.ts (hunk 4 above)
Comment (junbeomlee) [reply contracts/ethereum/tools/lib/utils/utils.ts]: (gene,owner) is the data to be signed!!

File: contracts/ethereum/packages/hatchout-cli/src/commands/ghost/list.ts (hunk 6 above)
Comment (junbeomlee) [reply contracts/ethereum/packages/hatchout-cli/src/commands/ghost/list.ts:7]: ok

File: contracts/ethereum/packages/hatchout-cli/src/commands/ghost/levelUp.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  async run() {
+    try {
+      const {args} = this.parse(LevelUp);
+      if (args.ghostIndex === undefined) {
+        this.error(`argument missing: ghostIndex`)
+      }
+
+      const hatchOut: HatchOut = HatchOut.createFromConfig();
+      const ghost: Ghost = await hatchOut.methods.getGhost(args.ghostIndex);
+      if (parseInt(ghost.level) > 2){
```
Comment (hihiboss) [inline contracts/ethereum/packages/hatchout-cli/src/commands/ghost/levelUp.ts:24]: I think that this logic has to be included in `hatchOut.methods.levelUp`.

**Pull request merged**